### DISCUSSION TITLE
How to manually construct and push spans to the batcher/exporter

### DISCUSSION THREAD

**Discussion by a10zn8** [open-telemetry/opentelemetry-go#4303] [Q&A]
Hello Everyone!

I have a specific setup, which includes a main server and several agent services located in various datacenters. The main server is responsible for creating spans and dispatching some requests to these agents. It sends the trace-id in the headers to these agents, and in turn, the agents record these spans and send them back as data. For me, connecting agents directly to the telemetry server is not possible.

Eventually, I end up with span data at the server, but I'm uncertain about the options available to construct a span from this data manually. The SpanProcessor interface accepts a trace.ReadOnlySpan, but implementation is not feasible due to its private method. I can construct tracepb.ResourceSpans, but there are no interfaces that accepts it. 

I'm seeking solutions to this problem. Are there alternative ways to create spans from raw data? Perhaps there might be some extras like custom builders or parsers available for different formats? In particular, I'm interested in building spans from the Zipkin JSON format.

Any guidance or suggestions would be highly appreciated. Thank you!

**Comment by MrAlias** [comment]
Thanks for the interest in the project.

The opentelemetry-go API and SDKs are platforms designed to generate and then transmit telemetry data. If you are looking to generate telemetry data in your own platform outside of opentelemetry-go you will need to handle the transmission of that data in your platform.

**Reply by a10zn8** [reply]
Thank you for your response.

Considering your explanation, I'm interested in knowing if there are any upcoming plans for OpenTelemetry Go to introduce an open API. This would ideally support the creation and manipulation of telemetry data externally, eliminating the need to duplicate already functioning code within the opentelemetry-go setup to make it work. Is there any possibility that such a feature is under consideration?

**Reply by MrAlias** [reply]
We do not have plans to support non-OpenTelemetry APIs with our SDKs or exporters.
